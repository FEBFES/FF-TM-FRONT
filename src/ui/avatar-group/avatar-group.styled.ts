import styled from 'styled-components';

export const SAvatarGroupContainer = styled.div`
  position: relative;
  display: flex;
  align-items: center;
`;

export const SAvatarCont = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const SOtherMember = styled.div`
  top: 50%;
  transform: translateY(-50%);
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 14px;
  font-weight: 600;
  position: absolute;
  background: var(--bg-0);
  border-radius: 100%;
  border: 1px solid var(--bg-border);
  width: 24px;
  height: 24px;
  color: var(--font-defautl);
`;
