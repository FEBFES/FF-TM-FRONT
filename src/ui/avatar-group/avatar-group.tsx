import React from 'react';
import { IMember } from '../../pages/KanbanPage/store/kanban.type';
import { Avatar, AvatarSizesType } from '../avatar/avatar';
import { Tooltip } from '../tooltip/tooltip';
import { PlacementType } from '../tooltip/tooltip';
import { getAvatarUrlOrHuman } from '../../utils/utils';
import {
  SAvatarGroupContainer,
  SAvatarCont,
  SOtherMember,
} from './avatar-group.styled';

interface AvatarGroupProps {
  members: IMember[];
  maxCount?: number;
  placement?: PlacementType;
  avatarSize: AvatarSizesType;
}

export const AvatarGroup: React.FC<AvatarGroupProps> = ({
  members,
  maxCount,
  avatarSize,
  placement,
}): JSX.Element => {
  const otherMember = members.length - (maxCount || 0);

  return (
    <SAvatarGroupContainer
      style={{
        marginRight: `-${10 * (maxCount ? maxCount : members.length)}px`,
      }}
    >
      {members.map((member: IMember, i: number) => {
        if (maxCount && i + 1 > maxCount) return null;
        return (
          <Tooltip
            key={member.id}
            placement={placement}
            title={`${member.username || member}`}
            style={{
              right: `${i * 10}px`,
              zIndex: `${i + 1}`,
            }}
          >
            <SAvatarCont key={`${i}-${member.id}`}>
              <Avatar
                size={avatarSize}
                src={getAvatarUrlOrHuman(member?.userPic)}
              />
            </SAvatarCont>
          </Tooltip>
        );
      })}
      {maxCount !== undefined && (
        <SOtherMember
          style={{
            left: `${maxCount * 14}px`,
            zIndex: maxCount + 1,
          }}
        >
          +{otherMember}
        </SOtherMember>
      )}
    </SAvatarGroupContainer>
  );
};
