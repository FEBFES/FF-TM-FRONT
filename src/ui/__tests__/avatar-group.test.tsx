import React from 'react';
import {AvatarGroup} from '../avatar-group/avatar-group';
import renderer from 'react-test-renderer';

it('SNAPSHOTS: UI - AvatarGroup component - render', () => {
    const members = [{
        displayName: 'DisplayName',
        email: 'test@mail.ru',
        firstName: 'Den',
        id: 123,
        lastName: 'Avabeloon',
        userPic: null,
        username: 'as',
        role: 'MEMBER'
    }]
    const elem = renderer.create(<AvatarGroup members={members} />).toJSON();
    expect(elem).toMatchSnapshot();
});