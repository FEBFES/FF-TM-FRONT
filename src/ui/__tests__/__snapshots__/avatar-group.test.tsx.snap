// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNAPSHOTS: UI - AvatarGroup component - render 1`] = `
<div
  className="sc-hLclGa cnUtay"
  style={
    Object {
      "marginRight": "-10px",
    }
  }
>
  <div
    className="sc-grXZZQ hBExFD"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "right": "0px",
        "zIndex": "1",
      }
    }
  >
    <div
      className="sc-grXZZQ hBExFD"
    >
      <div
        className="sc-eDnVMP iUtHUJ"
      >
        <img
          bordered={true}
          className="sc-bcPKhP djjAep"
          size="s"
          src="human.png"
        />
      </div>
    </div>
  </div>
</div>
`;
